File: packages/php-sdk/src/Types/Requests/Quote/UpdateCompanyRequest.php
<?php

declare(strict_types=1);

namespace TurboDocx\Types\Requests\Quote;

/**
 * Request for updating a company
 */
final class UpdateCompanyRequest
{
    public function __construct(
        public readonly ?string $name = null,
        public readonly ?string $phone = null,
        public readonly ?string $city = null,
        public readonly ?string $state = null,
        public readonly ?string $country = null,
        public readonly ?string $industryId = null,
    ) {}

    /**
     * @return array<string, mixed>
     */
    public function toArray(): array
    {
        $data = [];

        if ($this->name !== null) {
            $data['name'] = $this->name;
        }
        if ($this->phone !== null) {
            $data['phone'] = $this->phone;
        }
        if ($this->city !== null) {
            $data['city'] = $this->city;
        }
        if ($this->state !== null) {
            $data['state'] = $this->state;
        }
        if ($this->country !== null) {
            $data['country'] = $this->country;
        }
        if ($this->industryId !== null) {
            $data['industryId'] = $this->industryId;
        }

        return $data;
    }
}

File: packages/java-sdk/src/main/java/com/turbodocx/models/quote/UpdateCompanyRequest.java
package com.turbodocx.models.quote;

/**
 * Request to update a company.
 */
public class UpdateCompanyRequest extends TrackableRequest {
    private String name;
    private String phone;
    private String city;
    private String state;
    private String country;
    private String industryId;

    public String getName() { return name; }
    public void setName(String name) { this.name = name; markFieldSet("name"); }
    public String getPhone() { return phone; }
    public void setPhone(String phone) { this.phone = phone; markFieldSet("phone"); }
    public String getCity() { return city; }
    public void setCity(String city) { this.city = city; markFieldSet("city"); }
    public String getState() { return state; }
    public void setState(String state) { this.state = state; markFieldSet("state"); }
    public String getCountry() { return country; }
    public void setCountry(String country) { this.country = country; markFieldSet("country"); }
    public String getIndustryId() { return industryId; }
    public void setIndustryId(String industryId) { this.industryId = industryId; markFieldSet("industryId"); }
}
